src/test/java/org/kohsuke/github/GHEventPayloadTest.java

```diff
@@ -320,8 +323,32 @@ public void checkRunEvent() throws Exception {
                 .parseEventPayload(payload.asReader(), GHEventPayload.CheckRun.class);
         assertThat(event.getRepository().getName(), is("Hello-World"));
         assertThat(event.getAction(), is("created"));
-        assertThat(event.getCheckRun().getName(), is("Octocoders-linter"));
-        assertThat(event.getCheckRun().getHeadSha(), is("ec26c3e57ca3a959ca5aad62de7213c562f8c821"));
+
+        // Checks the deserialization of check_run
+        GHCheckRun checkRun = event.getCheckRun();
+        assertThat(checkRun.getName(), is("Octocoders-linter"));
+        assertThat(checkRun.getHeadSha(), is("ec26c3e57ca3a959ca5aad62de7213c562f8c821"));
+        assertThat(checkRun.getStatus(), is("completed"));
```

Comment:
@XiongKezhi I guess there are multiple states of the check run that need to be tested for, Is this the only one?

Reply:
Yeah, that's one thing we can do. Before, I just changed the original JSON in order to cover more methods for convenience, but later I found more sample responses [here](https://developer.github.com/v3/checks/runs/), they are much better and maybe we can use them later.